contrib/trace_propagators/src/main/java/io/opentelemetry/contrib/trace/propagation/B3PropagatorInjectorMultipleHeaders.java

```diff
@@ -32,7 +32,7 @@
     Objects.requireNonNull(setter, "setter");
 
     Span span = TracingContextUtils.getSpanWithoutDefault(context);
-    if (span == null) {
+    if (span == null || !span.getContext().isValid()) {
```

Comment:
Wouldn't the code be easier if you replace `TracingContextUtils.getSpanWithoutDefault` with `TracingContextUtils.getSpan` and then drop null check and use only `isValid`?

Reply:
Yeah, I was thinking about it too. I guess I prefer to do a simple `null` check before doing a full check, i.e. `getContext()` + `isValid()` ;)

FWIW though, I'm thinking, after cases like these, to remove the `getSpanWithoutDefault()` overload (which I might end up doing after doing some context-prop updates).

Reply:
There might be propagators that do will be able to meaninfully inject something only based on TraceState, even for invalid trace and span IDs. So while an utility function would be good, I think the final decision should be left to the propagator.

Reply:
> There might be propagators that will be able to meaningfully inject something only based on TraceState

Right, exactly, I had totally forgotten about this case :)

Reply:
The w3c standard says that traceparent must be present so I am worried that if that can be the case to only propagate tracestate

Reply:
Sure, this does not apply to W3C but maybe other propagators (vendor ones for example).